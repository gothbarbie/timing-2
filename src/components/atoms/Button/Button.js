import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Default = styled.button`
  cursor: pointer;
  width: ${({ size }) => size === 's' && 'auto'};
  width: ${({ fullWidth }) => fullWidth && '100%'};

  background-color: transparent;
  color: ${({ theme }) => theme.palette.grayScale[0]};
  color: ${({ active, theme }) => active && theme.palette.primary[0]};
  outline: none;
  font-size: 1.6rem;
  font-weight: 500;
  border: 1px solid transparent;
  border-color: ${({ active, theme }) => active && theme.palette.grayScale[0]};
  padding: 1rem 2rem;
  padding: ${({ size }) => size === 's' && '.5rem'};

  &:focus {
    border-color: ${({ theme }) => theme.palette.primary[0]};
  }

  &:hover {
    background-color: ${({ theme }) => theme.palette.grayScale[1]};
    color: ${({ theme }) => theme.palette.primary[0]};
    border-color: ${({ theme }) => theme.palette.grayScale[0]};
  }

  &:active {
    box-shadow: inset 0 2px 2px rgba(0, 0, 0, 0.25);
  }

  border-radius: 4px;
`

const WithIcon = styled(Default)`
  &:focus {
    border-color: ${({ theme }) => theme.palette.grayScale[2]};
    color: ${({ theme }) => theme.palette.primary[0]};
  }

  &:hover {
    border-color: ${({ theme }) => theme.palette.grayScale[0]};
    color: ${({ theme }) => theme.palette.primary[0]};
    fill: ${({ theme }) => theme.palette.primary[0]};
  }
`

const Primary = styled(Default)`
  background-color: ${({ theme }) => theme.palette.primary[0]};
  color: ${({ theme }) => theme.palette.white[0]};

  box-shadow: ${({ active }) =>
    active ? 'inset 0 2px 2px rgba(0, 0, 0, 0.25)' : 'none'};

  &:focus {
    border-color: ${({ theme }) => theme.palette.white[0]};
  }

  &:hover {
    background-color: ${({ theme }) => theme.palette.primary[0]};
    border-color: ${({ theme }) => theme.palette.white[0]};
    color: ${({ theme }) => theme.palette.white[0]};
  }
`

export const Button = ({
  active = false,
  children,
  fullWidth,
  hasIcon,
  primary,
  size,
  title,
  onClick,
}) => {
  if (primary)
    return (
      <Primary
        active={active}
        fullWidth={fullWidth}
        onClick={onClick}
        size={size}
        title={title}
      >
        {children}
      </Primary>
    )
  if (hasIcon) {
    return (
      <WithIcon
        active={active}
        fullWidth={fullWidth}
        hasIcon={hasIcon}
        onClick={onClick}
        title={title}
      >
        {children}
      </WithIcon>
    )
  }

  return (
    <Default
      active={active}
      fullWidth={fullWidth}
      onClick={onClick}
      size={size}
      title={title}
    >
      {children}
    </Default>
  )
}

Button.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.node,
  fullWidth: PropTypes.bool,
  hasIcon: PropTypes.bool,
  primary: PropTypes.bool,
  size: PropTypes.string,
  title: PropTypes.string,
  onClick: PropTypes.func,
}

/** @component */
export default Button
