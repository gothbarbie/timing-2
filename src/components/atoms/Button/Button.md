# Button

```js
  <Button>Test Button</Button>
```
