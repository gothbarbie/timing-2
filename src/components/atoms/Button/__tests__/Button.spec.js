import React from 'react'
import { shallow } from 'enzyme'

import Button from '../Button'

describe('components/atoms/Button', () => {
  it('renders Default style', () => {
    const component = shallow(<Button>Text</Button>)

    expect(component).toMatchSnapshot()
  })

  it('renders Primary style', () => {
    const component = shallow(<Button primary>Text</Button>)

    expect(component).toMatchSnapshot()
  })

  it('renders Icon style', () => {
    const component = shallow(
      <Button hasIcon>
        <i />
      </Button>
    )

    expect(component).toMatchSnapshot()
  })
})
