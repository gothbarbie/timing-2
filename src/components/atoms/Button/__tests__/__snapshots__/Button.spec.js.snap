// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/Button renders Default style 1`] = `
<Default
  active={false}
>
  Text
</Default>
`;

exports[`components/atoms/Button renders Icon style 1`] = `
<WithIcon
  active={false}
  hasIcon={true}
>
  <i />
</WithIcon>
`;

exports[`components/atoms/Button renders Primary style 1`] = `
<Primary
  active={false}
>
  Text
</Primary>
`;
